### PR TITLE
Added improvements to Github workflow

### PR DESCRIPTION
**Pull request by marchingon12**
This pull request introduces several enhancements to streamline our GitHub workflow and improve project contribution guidelines:

1. **Contributing Guidelines**: Added a `CONTRIBUTING.md` file outlining the contribution process for this project. Contributors are encouraged to refer to this document when making contributions via pull requests.

2. **Updated README**: Enhanced the README with the following:
- **Star History Diagram**: Included a visual representation of star history to provide insight into project popularity and growth.
- **Username Update**: Updated references to the project owner's username.
- **Contribution Reference**: Included instructions in the README directing contributors to refer to `CONTRIBUTING.md` for guidance on making contributions via pull requests.

3. **Pull Request Templates**: Added two pull request templates to standardize the PR submission process:
- **General PR Template**: For general pull requests, bug fixes, and feature additions.
- **Translation PR Template**: Specifically tailored for translation-related pull requests.

These improvements aim to facilitate a smoother contribution process and enhance project documentation for both existing contributors and newcomers.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -97,14 +101,18 @@ It is easiest to make a translation using the Android Studio XML editor, but you
 - [Dagger-Hilt](https://dagger.dev/hilt) For [Dependency injection (DI)](https://developer.android.com/training/dependency-injection)
 - [Room database](https://developer.android.com/jetpack/androidx/releases/room) - Persistence library provides an abstraction layer over SQLite to allow for more robust database access while harnessing the full power of SQLite.
```
Comment (starry-shivam) [inline README.md:103]: Also add a divider `------` between tech-stack and star history

File: .github/PULL_REQUEST_TEMPLATE/pull_request_template.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+<!--- Please link to the issue here: -->
+
+## Type of change 
+<!--- Is this a bug fix, adding a feature... -->
+
+### Pull Request checklist
+<!-- Before submitting the PR, please address each item. Use [x] to check the boxes -->
+- [ ] The commit message follows our guidelines
+- [ ] Made sure that your PR is not duplicate
+- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not (for bug fixes/features). Your PR should pass all CI checks in our Gtihub Actions [Workflow](https://github.com/Pool-Of-Tears/GreenStash/actions)
```
[the remark this answers is not in the document]
Comment (marchingon12) [reply .github/PULL_REQUEST_TEMPLATE/pull_request_template.md:17]: > - [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not (for bug fixes/features). Your PR should pass all CI checks in our Gtihub Actions [Workflow](https://github.com/Pool-Of-Tears/GreenStash/actions)

Basically asking whether the PR has gone through tests, and if not, explain why.

[the remark this answers is not in the document]
Comment (starry-shivam) [reply .github/PULL_REQUEST_TEMPLATE/pull_request_template.md:17]: Oh got it

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+**We'll probably ignore and close your issue if it's not using the existing templates or doesn't contain sufficient description.**
+
+## Bug Report
+
+When submitting a bug report, please make sure your issue contains **enough** information for reproducing the problem, including the options or the custom command being used, the link to the video, and other fields in the issue template.
+
+## Feature Request
+
+**GreenStash** is designed to provide users with a straightforward and effective tool for managing their finances on Android. Our focus is to offer essential budgeting functionalities in an intuitive interface, helping users track their expenses and savings effortlessly. As such, we prioritize feature requests that align with our core mission and the fundamental principles of budget management.
```
[the remark this answers is not in the document]
Comment (marchingon12) [reply CONTRIBUTING.md]: Please check again to see if description matches the requested change.

[the remark this answers is not in the document]
Comment (starry-shivam) [reply CONTRIBUTING.md]: Seems good to me now

**Pull request merged**